WordPress/Classes/ViewRelated/Reader/Detail/ReaderDetailViewController.swift

```diff
@@ -909,16 +900,20 @@ extension ReaderDetailViewController: UIScrollViewDelegate {
         // Using `safeAreaLayoutGuide.layoutFrame.height` because it doesn't
         // change when we extend the scroll view size by hiding the toolbar
         if (currentOffset + view.safeAreaLayoutGuide.layoutFrame.height) > likesContainerView.frame.minY {
-            toolbarHiddenDebouncer.send(false) // Reached bottom (controls, comments, etc)
+            setNeedsToolbarHidden(false) // Reached bottom (controls, comments, etc)
         } else if currentOffset > lastContentOffset && currentOffset > 0 {
-            toolbarHiddenDebouncer.send(true) // Scrolling down
+            setNeedsToolbarHidden(true) // Scrolling down
         } else if currentOffset < lastContentOffset {
-            toolbarHiddenDebouncer.send(false) // Scrolling up
+            setNeedsToolbarHidden(false) // Scrolling up
         }
         lastContentOffset = currentOffset
         layoutHeroView()
     }
 
+    // In the `scrollViewDidScroll` function, "toolbar hidden" is toggled repeatedly when the scroll view in the
+    // middle of scrolling animation. This debouncer is introduced to avoid toggling
+    // `navigationController.toolbarHidden`, which causes inifity loops.
+    // Sentry issue: https://a8c.sentry.io/issues/6884521550
     private func setNeedsToolbarHidden(_ isHidden: Bool) {
```

Comment:
I'm not sure if the timer-based implementation is better than the publisher-based implementation.

The implementation here does not actually "debounce". It just delays the actual `setToolbarHidden` call until no one calls it again in 0.25 seconds. This implementation gives you the UX where the toolbar becomes hidden/visible when the scroll view stops scrolling. If that's the UX we want, I don't think we should use this timer here. We can implement the "didEndScroll" and "didEndDecelerating" delegate methods.